### ISSUE TITLE
Starting a podman (Run button in Dashboard) when there is something broken shows Error: [object Object] instead of proper message

### ISSUE THREAD

**Issue by odockal** [containers/podman-desktop#3913]
### Bug description
I am not sure how I got into this state, seems like podman machine was not initialized correctly.

![Runpodman-error-object](https://github.com/containers/podman-desktop/assets/19164299/13df674d-5a4f-4ae4-9e9c-4cf226de465f)

### Operating system

Windows 10

### Installation Method

Other

### Version

1.4.0

### Steps to reproduce

1. Have erroneous state of WSL (I do not know how, but I got a "Catastrophic failure" during `wsl --install --no-distribution`)
2. Install podman
3. Initialize and run Podman
4. Podman machine is started
Actual: Podman is not started, error message is not properly shown, errors in console log.

I am aware that reproduce this issue is gonna be hard, but maybe console errors would help identify the place.

### Relevant log output

```shell
main ↪️ [podman] Objectcancelled: falsecommand: "podman.exe"exitCode: 125killed: falsemessage: "Command execution failed with exit code 125"name: "Command execution failed with exit code 125"stderr: "Error: the WSL bootstrap script failed: exit status 0xffffffff"stdout: "Starting machine \"podman-machine-default\" [[Prototype]]: Object
console.<computed> @ VM5:2
-------------------------------------------------------------
PreferencesConnectionActions.svelte:48 Error: [object Object]
u	@	PreferencesConnectionActions.svelte:48
------------------------------------------------------------
Error while starting the provider Error: [object Object]
u @ ProviderConfigured.svelte:38
```

**Comment by datho7561** [comment]
I ran into a similar issue when installing and starting `podman` through Podman Desktop. I ran `wsl --install --no-distribution` before launching. I opened the Chrome logs and got:

![Screenshot 2023-10-06 114627](https://github.com/containers/podman-desktop/assets/22376627/d1ff1f15-e596-4c82-b950-d1acda8dc369)

**Comment by datho7561** [comment]
In my case, something was wrong with the podman machine, deleting and recreating from within the podman desktop settings solved the issue that caused the error popup.

**Comment by Pyroglyph** [comment]
Mine initially worked, but after installing Compose through Podman Desktop it started doing this.

**Comment by dgolovin** [comment]
`Error: [object Object]` should be fixed in main branch, because #4231 fix was merged.

**Comment by lstocchi** [comment]
I was able to verify it works fine now. Thanks @dgolovin . One less issue to work on for me 🍻 😆
